src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/UniqueAccessCodeServiceImpl.java

```diff
@@ -369,31 +366,9 @@ private UniqueAccessCodeDTO createUniqueAccessCodeDTO(
    */
   private CollectionCase findValidNonHICase(String uprnAsString, String uacHash)
       throws CTPException {
-    List<CollectionCase> cases = dataRepo.readCollectionCasesByUprn(uprnAsString);
-
-    CollectionCase newestPrimaryCase = null;
-    if (cases.size() >= 1) {
-      // filter out just the non HI which are address valid and sort in ascending order by
-      // createdDateTime
-      List<CollectionCase> addressValidNonHICases =
-          cases
-              .stream()
-              .sorted(Comparator.comparing(CollectionCase::getCreatedDateTime))
-              .filter(c -> (!c.getCaseType().equals(CaseType.HI.name())))
-              .filter(c -> (!c.isAddressInvalid()))
-              .collect(Collectors.toList());
-      if (addressValidNonHICases.size() >= 1) {
-        // take the case last in the list ie the newest by createdDateTime
-        newestPrimaryCase = addressValidNonHICases.get(addressValidNonHICases.size() - 1);
-        if (addressValidNonHICases.size() > 1) {
-          // carry on with the latest case but log a warning as it indicates RM events have got
-          // out of step
-          log.with("UACHash", uacHash)
-              .with("UPRN", uprnAsString)
-              .warn("More than one active case found for UPRN");
-        }
-      }
-    }
-    return newestPrimaryCase;
+
+    Optional<CollectionCase> caseFound =
```

Comment:
From a consistency perspective - this is to here to avoid dealing with Optional<CollectionCase>, but the caller in this same class is happy to deal with Optional<UAC>?

Look at the caller to see what I mean.

Which is it here, null or Optional, Optional or null?